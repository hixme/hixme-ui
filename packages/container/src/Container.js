import React from 'react'
import PropTypes from 'prop-types'
import ContainerBase from './ContainerBase'

const propTypes = {
  height: PropTypes.string,
  maxHeight: PropTypes.string,
  maxWidth: PropTypes.string,
  minHeight: PropTypes.string,
  minWidth: PropTypes.string,
  width: PropTypes.string,

  margin: PropTypes.string,
  marginLeft: PropTypes.string,
  marginRight: PropTypes.string,
  marginTop: PropTypes.string,
  marginBottom: PropTypes.string,
  marginSides: PropTypes.string,

  padding: PropTypes.string,
  paddingLeft: PropTypes.string,
  paddingRight: PropTypes.string,
  paddingTop: PropTypes.string,
  paddingBottom: PropTypes.string,
  paddingSides: PropTypes.string,
  withPadding: PropTypes.bool,

  border: PropTypes.string,
  borderColor: PropTypes.string,
  dashed: PropTypes.bool,
  rounded: PropTypes.bool,

  default: PropTypes.bool,
  primary: PropTypes.bool,
  success: PropTypes.bool,
  warning: PropTypes.bool,
  error: PropTypes.bool,
  muted: PropTypes.bool,

  white: PropTypes.bool,
  black: PropTypes.bool,
  red: PropTypes.bool,
  blue: PropTypes.bool,
  purple: PropTypes.bool,
  yellow: PropTypes.bool,
  green: PropTypes.bool,

  lighter: PropTypes.bool,
  light: PropTypes.bool,
  lightest: PropTypes.bool,

  textLeft: PropTypes.bool,
  textCenter: PropTypes.bool,
  textRight: PropTypes.bool,

  shadow: PropTypes.bool,
  heavyShadow: PropTypes.bool,
  animate: PropTypes.bool,

  flex: PropTypes.bool,
  flexInline: PropTypes.bool,
  flexRow: PropTypes.bool,
  flexColumn: PropTypes.bool,
  flexWrap: PropTypes.bool,
  justifyStart: PropTypes.bool,
  justifyEnd: PropTypes.bool,
  justifyCenter: PropTypes.bool,
  justifySpaceBetween: PropTypes.bool,
  justifySpaceAround: PropTypes.bool,
  alignStart: PropTypes.bool,
  alignEnd: PropTypes.bool,
  alignCenter: PropTypes.bool,
  alignSpaceBetween: PropTypes.bool,
  alignSpaceAround: PropTypes.bool,

  direction: PropTypes.string,
  alignItems: PropTypes.string,
  alignContent: PropTypes.string,
  justifyContent: PropTypes.string,
  alignSelf: PropTypes.string,
  flexBasis: PropTypes.string,
  flexGrow: PropTypes.number,
  flexShrink: PropTypes.number,

  withOutline: PropTypes.bool,
}

const defaultProps = {
  height: null,
  maxHeight: null,
  maxWidth: null,
  minHeight: null,
  minWidth: null,
  width: null,

  margin: null,
  marginLeft: null,
  marginRight: null,
  marginTop: null,
  marginBottom: null,
  marginSides: null,

  padding: '20px',
  paddingLeft: '',
  paddingRight: '',
  paddingTop: '',
  paddingBottom: '',
  paddingSides: '',
  withPadding: false,

  border: 'none',
  borderColor: '',
  dashed: false,
  rounded: false,

  default: false,
  primary: false,
  success: false,
  warning: false,
  error: false,
  muted: false,

  white: false,
  black: false,
  red: false,
  blue: false,
  purple: false,
  yellow: false,
  green: false,

  lighter: false,
  light: false,
  lightest: false,

  textLeft: false,
  textCenter: false,
  textRight: false,

  shadow: false,
  heavyShadow: false,
  animate: false,

  flex: false,
  flexInline: false,
  flexRow: false,
  flexColumn: false,
  flexWrap: false,
  justifyStart: false,
  justifyEnd: false,
  justifyCenter: false,
  justifySpaceBetween: false,
  justifySpaceAround: false,
  alignStart: false,
  alignEnd: false,
  alignCenter: false,
  alignSpaceBetween: false,
  alignSpaceAround: false,

  direction: '',
  alignItems: '',
  alignContent: '',
  justifyContent: '',
  alignSelf: '',
  flexBasis: '',
  flexGrow: null,
  flexShrink: null,

  withOutline: false,
}

const Container = props => <ContainerBase {...props} />

Container.displayName = 'hui:Container'

Container.huiName = 'Container'

Container.propTypes = propTypes

Container.defaultProps = defaultProps

export default Container
