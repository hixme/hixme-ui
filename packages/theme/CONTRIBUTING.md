# Contributing

Open a PR at https://github.com/hixme/hixme-ui.
