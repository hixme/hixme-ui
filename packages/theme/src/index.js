export const colors = {
  white: 'white',
  black: 'black',
  grey30: '#3A4951',
  grey80: '#8F9DA8',
  grey100: '#AAB5BD',
  grey110: '#B9C5CD',
  grey120: '#CBD6DB',
  grey130: '#f7fcff',
  teal: '#2EAE8F',
  blue: '#3997E2',
  red: '#EB5565',
  yellow: '#E0BC4D',
  purple: '#A374D2',
  green: '#88b963',
}

const gradients = {
  teal: 'linear-gradient(180deg, #20BCA1 0%, #159777 100%)',
  red: 'linear-gradient(0deg, #DE293C 0%, #FF7272 100%)',
  yellow: 'linear-gradient(-1deg, #FF8C00 0%, #F3C200 100%)',
  blue: 'linear-gradient(-1deg, #2583C4 0%, #30A0EE 100%)',
  grey: 'linear-gradient(180deg, #CCD2D6 0%, #a4adb3 100%)',
}

const fontSizes = {
  jumbo: '22px',
  larger: '20px',
  large: '18px',
  medium: '16px',
  default: '14px',
  small: '12px',
  smaller: '10px',
  smallest: '8px',
}

const fontWeights = {
  default: 'normal',
}

const titleFontSizes = {
  jumbo: '38px',
  larger: '36px',
  large: '34px',
  medium: '32px',
  default: '30px',
  small: '28px',
  smaller: '26px',
  smallest: '24px',
}

const buttonFontSizes = {
  mini: fontSizes.small,
  small: fontSizes.default,
  large: fontSizes.large,
  jumbo: fontSizes.jumbo,
}

const buttonHeights = {
  mini: '28px',
  small: '36px',
  default: '42px',
  large: '50px',
  jumbo: '65px',
}

const buttonMinWidths = {
  mini: '80px',
  small: '80px',
  default: '100px',
  large: '100px',
  jumbo: '100px',
}

const buttonTextColors = {
  default: colors.white,
}

const buttonGradientColors = {
  primary: gradients.teal,
  secondary: gradients.grey,
  default: gradients.teal,
  red: gradients.red,
  danger: gradients.red,
  error: gradients.red,
  blue: gradients.blue,
  gold: gradients.yellow,
}

const shadows = {
  medium: '0px 5px 9px 0px rgba(56,59,60,0.22);',
  default: '0px 2px 6px 0px rgba(56,59,60,0.18);',
  none: '0px 0px 0px 0px rgba(0,0,0,0);',
}

const buttonBorderRadius = {
  none: '0',
  default: '4px',
}

/* eslint-disable max-len */
export const fontFamily = {
  thinnest: '\'Avenir-Next_UltraLight\', \'HelveticaNeue-UltraLight\', \'Helvetica Neue Ultra Light\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  thinner: '\'Avenir-Next_Light\', \'HelveticaNeue-Thin\', \'Helvetica Neue Thin\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  thin: '\'Avenir-Next_Regular\', \'HelveticaNeue-Light\', \'Helvetica Neue Light\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  normal: '\'Avenir-Next_Medium\', \'HelveticaNeue-Roman\', \'Helvetica Neue Roman\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  bold: '\'Avenir-Next_Demi\', \'HelveticaNeue-Medium\', \'Helvetica Neue Medium\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  bolder: '\'Avenir-Next_Bold\', \'HelveticaNeue-Bold\', \'Helvetica Neue Bold\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  boldest: '\'Avenir-Next_Heavy\', \'HelveticaNeue-Heavy\', \'Helvetica Neue Heavy\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
}

const defaults = {
  gradient: gradients.teal,
  fontSize: fontSizes.default,
  buttonHeight: buttonHeights.default,
  textColor: colors.grey30,
  fontFamily: fontFamily.normal,
  fontWeight: fontWeights.default,
  backgroundColor: colors.white,
  borderColor: colors.grey110,
  titleFontSize: titleFontSizes.default,
}

export const colorTheme = {
  default: colors.grey30,
  primary: colors.teal,
  success: colors.teal,
  warning: colors.yellow,
  error: colors.red,
  danger: colors.red,
  muted: colors.grey80,

  white: colors.white,
  black: colors.black,
  red: colors.red,
  blue: colors.blue,
  purple: colors.purple,
  yellow: colors.yellow,
  green: colors.green,

  lighter: colors.grey110,
  light: colors.grey80,
  lightest: colors.grey130,
}

const outlineButtonColors = {
  ...colorTheme,
  default: colors.grey110,
}

const theme = {
  colors: colorTheme,
  backgroundColors: {
    ...colorTheme,
    default: defaults.backgroundColor,
  },
  borderColors: {
    ...colorTheme,
    default: defaults.borderColor,
  },
  buttonTextColors,
  buttonGradientColors,
  buttonFontSizes,
  buttonHeights,
  buttonMinWidths,
  buttonBorderRadius,
  outlineButtonColors,
  defaults,
  fontSizes,
  fontFamily,
  fontWeights,
  gradients,
  shadows,
  textColors: {
    ...colorTheme,
    default: defaults.textColor,
  },
  titleFontSizes,
}

export default theme

export ThemeProvider from './ThemeProvider'
