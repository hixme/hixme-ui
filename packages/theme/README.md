# hixme-ui theme

```bash
npm i --save @hixme-ui/theme
```

## Theme

HUI theme colors and defaults

```javascript
import theme from '@hixme-ui/theme

const primary = theme.colors.primary
```


## ThemeProvider

HUI theme provider, which is just an implementation of styled-components
ThemeProvider with the HUI theme as the default

```javascript
import { ThemeProvider } from '@hixme-ui/theme'

const App = ({ children }) => (
  <ThemeProvider>
    {children}
  </ThemeProvider>
)
```
