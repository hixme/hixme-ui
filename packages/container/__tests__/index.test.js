import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from '@hixme-ui/theme'
import 'jest-styled-components'

import Container from '../src/index'

describe('Container', () => {
  test('it renders', () => {
    const component = renderer.create(
      <ThemeProvider><Container /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree.type).toBe('div')
  })

  test('it has 20px padding by default', () => {
    const component = renderer.create(
      <ThemeProvider><Container /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding', '20px')
  })

  test('it renders with a 1px solid red outline when withOutline is true', () => {
    const component = renderer.create(
      <ThemeProvider><Container withOutline /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('outline', '1px solid red')
  })

  test('it should contain 30px padding and margin: 2px 8px 3px 7px', () => {
    const component = renderer.create(
      <ThemeProvider><Container padding='30px' margin='2px 8px 3px 7px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding', '30px')
    expect(tree).toHaveStyleRule('margin', '2px 8px 3px 7px')
  })

  test('it should render with 80px left and right margins and padding', () => {
    const component = renderer.create(
      <ThemeProvider><Container paddingSides='80px' marginSides='80px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin-left', '80px')
    expect(tree).toHaveStyleRule('margin-right', '80px')
    expect(tree).toHaveStyleRule('padding-left', '80px')
    expect(tree).toHaveStyleRule('padding-right', '80px')
  })

  test('it should render with left padding of 30px', () => {
    const component = renderer.create(
      <ThemeProvider><Container padding='10px' paddingLeft='30px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding-left', '30px')
  })

  test('it should render with right padding of 90px', () => {
    const component = renderer.create(
      <ThemeProvider><Container padding='10px' paddingRight='90px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding-right', '90px')
  })

  test('it should render with bottom padding of 16px', () => {
    const component = renderer.create(
      <ThemeProvider><Container padding='10px' paddingBottom='16px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding-bottom', '16px')
  })

  test('it should render with top padding of 36px', () => {
    const component = renderer.create(
      <ThemeProvider><Container padding='10px' paddingTop='36px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding-top', '36px')
  })

  test('it should render with left margin of 30px', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='10px' marginLeft='30px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin-left', '30px')
  })

  test('it should render with right margin of 90px', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='10px' marginRight='90px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin-right', '90px')
  })

  test('it should render with bottom margin of 16px', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='10px' marginBottom='16px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin-bottom', '16px')
  })

  test('it should render with top margin of 36px', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='10px' marginTop='36px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin-top', '36px')
  })

  test('it should contain 100px width and 200px height', () => {
    const component = renderer.create(
      <ThemeProvider><Container width='100px' height='200px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('width', '100px')
    expect(tree).toHaveStyleRule('height', '200px')
  })

  test('it should contain a min-width of 20px and max-width of 100px', () => {
    const component = renderer.create(
      <ThemeProvider><Container minWidth='20px' maxWidth='100px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('min-width', '20px')
    expect(tree).toHaveStyleRule('max-width', '100px')
  })

  test('it should contain a min-height of 20px and max-height of 100px', () => {
    const component = renderer.create(
      <ThemeProvider><Container minHeight='20px' maxHeight='100px' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('min-height', '20px')
    expect(tree).toHaveStyleRule('max-height', '100px')
  })

  test('it should contain text-align: left', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='20px' textAlign='left' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin', '20px')
    expect(tree).toHaveStyleRule('text-align', 'left')
  })

  test('it should contain text-align: left again', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='20px' textLeft /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin', '20px')
    expect(tree).toHaveStyleRule('text-align', 'left')
  })

  test('it should contain text-align: center again', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='20px' textCenter /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin', '20px')
    expect(tree).toHaveStyleRule('text-align', 'center')
  })

  test('it should contain text-align: right', () => {
    const component = renderer.create(
      <ThemeProvider><Container margin='20px' textRight /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('margin', '20px')
    expect(tree).toHaveStyleRule('text-align', 'right')
  })

  test('it should render with border: 130px solid purple', () => {
    const component = renderer.create(
      <ThemeProvider><Container border='130px solid purple' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border', '130px solid purple')
  })

  test('it should render with 6px border radius', () => {
    const component = renderer.create(
      <ThemeProvider><Container rounded /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border-radius', '6px')
  })

  test('it should render with 2 pixel dashed border', () => {
    const component = renderer.create(
      <ThemeProvider><Container dashed /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border', '2px dashed')
  })

  test('it should render a red border-color', () => {
    const component = renderer.create(
      <ThemeProvider><Container borderColor='red' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border-color', 'red')
  })

  test('it should render a green bottom border', () => {
    const component = renderer.create(
      <ThemeProvider><Container borderBottom='1px solid green' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border-bottom', '1px solid green')
  })

  test('it should render with background: black', () => {
    const component = renderer.create(
      <ThemeProvider><Container background='black' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('background', 'black')
  })

  test('it should render with display: flex', () => {
    const component = renderer.create(
      <ThemeProvider><Container flex /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('display', 'flex')
  })

  test('it should render with justify-content: flex-start', () => {
    const component = renderer.create(
      <ThemeProvider><Container flex justifyStart /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('display', 'flex')
    expect(tree).toHaveStyleRule('justify-content', 'flex-start')
  })

  test('it should render with align-items: center', () => {
    const component = renderer.create(
      <ThemeProvider><Container flex alignCenter /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('display', 'flex')
    expect(tree).toHaveStyleRule('align-items', 'center')
  })

  test('it should render with flex-direction: column', () => {
    const component = renderer.create(
      <ThemeProvider><Container flexColumn /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('display', 'flex')
    expect(tree).toHaveStyleRule('flex-direction', 'column')
  })

  test('it renders with no padding when noPadding is true', () => {
    const component = renderer.create(
      <ThemeProvider><Container noPadding /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding', '0')
  })

  test('it should render with opacity .6', () => {
    const component = renderer.create(
      <ThemeProvider><Container opacity='.6' /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('opacity', '.6')
  })
})
