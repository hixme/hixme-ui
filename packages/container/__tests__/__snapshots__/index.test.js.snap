// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container it has 20px padding by default 1`] = `
.c0 {
  background: none;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it renders with a 1px solid red outline when withOutline is true 1`] = `
.c0 {
  background: none;
  padding: 20px;
  border: none;
  outline: 1px solid red;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it renders with no padding when noPadding is true 1`] = `
.c0 {
  background: none;
  padding: 0;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain 30px padding and margin: 2px 8px 3px 7px 1`] = `
.c0 {
  background: none;
  padding: 30px;
  margin: 2px 8px 3px 7px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain 100px width and 200px height 1`] = `
.c0 {
  background: none;
  width: 100px;
  height: 200px;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height="200px"
  muted={false}
  width="100px"
/>
`;

exports[`Container it should contain a min-height of 20px and max-height of 100px 1`] = `
.c0 {
  background: none;
  max-height: 100px;
  min-height: 20px;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain a min-width of 20px and max-width of 100px 1`] = `
.c0 {
  background: none;
  max-width: 100px;
  min-width: 20px;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain text-align: center again 1`] = `
.c0 {
  background: none;
  text-align: center;
  padding: 20px;
  margin: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain text-align: left 1`] = `
.c0 {
  background: none;
  text-align: left;
  padding: 20px;
  margin: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain text-align: left again 1`] = `
.c0 {
  background: none;
  text-align: left;
  padding: 20px;
  margin: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should contain text-align: right 1`] = `
.c0 {
  background: none;
  text-align: right;
  padding: 20px;
  margin: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render a red border-color 1`] = `
.c0 {
  background: none;
  padding: 20px;
  border: none;
  border-color: red;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with 2 pixel dashed border 1`] = `
.c0 {
  background: none;
  padding: 20px;
  border: none;
  border: 2px dashed;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with 6px border radius 1`] = `
.c0 {
  background: none;
  padding: 20px;
  border: none;
  border-radius: 6px;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with 80px left and right margins and padding 1`] = `
.c0 {
  background: none;
  padding: 20px;
  padding-left: 80px;
  padding-right: 80px;
  margin-left: 80px;
  margin-right: 80px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with align-items: center 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with background: black 1`] = `
.c0 {
  background: black;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with background: white 1`] = `
.c0 {
  background: white;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={true}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with border: 130px solid purple 1`] = `
.c0 {
  background: none;
  padding: 20px;
  border: 130px solid purple;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with bottom margin of 16px 1`] = `
.c0 {
  background: none;
  padding: 20px;
  margin: 10px;
  margin-bottom: 16px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with bottom padding of 16px 1`] = `
.c0 {
  background: none;
  padding: 10px;
  padding-bottom: 16px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with display: flex 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with flex-direction: column 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with justify-content: flex-start 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 20px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with left margin of 30px 1`] = `
.c0 {
  background: none;
  padding: 20px;
  margin: 10px;
  margin-left: 30px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with left padding of 30px 1`] = `
.c0 {
  background: none;
  padding: 10px;
  padding-left: 30px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with right margin of 90px 1`] = `
.c0 {
  background: none;
  padding: 20px;
  margin: 10px;
  margin-right: 90px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with right padding of 90px 1`] = `
.c0 {
  background: none;
  padding: 10px;
  padding-right: 90px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with top margin of 36px 1`] = `
.c0 {
  background: none;
  padding: 20px;
  margin: 10px;
  margin-top: 36px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;

exports[`Container it should render with top padding of 36px 1`] = `
.c0 {
  background: none;
  padding: 10px;
  padding-top: 36px;
  border: none;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
/>
`;
